acceptance_tests/features/steps/notification_letter.py

```diff
@@ -64,24 +66,35 @@ def _get_sftp_client():
 
 
 @retry(retry_on_exception=lambda e: isinstance(e, FileNotFoundError),
-       wait_fixed=1000, stop_max_attempt_number=120)
-def _get_path_of_latest_notification_file(client, start_of_test, survey_ref, period):
-    logger.info('Loading file from SFTP')
-    files = _get_files_ordered_by_modified_time_desc(client)
-    if not files:
+       wait_fixed=5000, stop_max_attempt_number=24)
+def _check_notification_files_have_iacs(client, start_of_test, survey_ref, period, expected_iacs):
+    logger.info('Checking for files on SFTP server')
+    files = _get_files_filtered_by_name_and_modified_time(client, survey_ref, period, start_of_test)
+    if len(files) == 0:
         raise FileNotFoundError
 
-    latest_file_attributes = files[0]
-    filename = latest_file_attributes.filename
-    if f'{survey_ref}_{period}' not in filename:
-        raise FileNotFoundError
+    seen_iacs = []
+
+    for file in files:
+        file_path = f'{Config.SFTP_DIR}/{file.filename}'
 
-    modified_time_of_file = datetime.fromtimestamp(latest_file_attributes.st_mtime)
-    start_of_test = _round_to_minute(start_of_test)  # Time on SFTP on CI is only precise to the minute
-    if start_of_test > modified_time_of_file:
+        with client.open(file_path) as sftp_file:
+            content = str(sftp_file.read())
+
+            for iac in expected_iacs:
+                if iac in content:
+                    seen_iacs.append(iac)
+
+    if set(seen_iacs) != set(expected_iacs):
+        file_names = [f.filename for f in files]
+        logger.info('Unable to find all iacs', files_found=file_names, seen_iacs=seen_iacs, expected_iacs=expected_iacs)
         raise FileNotFoundError
 
-    return f'{Config.SFTP_DIR}/{filename}'
+    for file in files:
```

Comment:
This is potentially problematic. We don't know _which_ files the IAC codes were in, and we could end up deleting files which didn't have *any* IAC codes we expected in them, which might hamper test debugging

Reply:
Okay, is removing the delete the files here and 'reseting' the sftp server at the start of the ATs better?